tests/scripts/all.sh

```diff
@@ -2125,14 +2125,21 @@ component_test_everest () {
 
 component_test_everest_curve25519_only () {
     msg "build: Everest ECDH context, only Curve25519" # ~ 6 min
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
     scripts/config.py set MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED
     scripts/config.py unset MBEDTLS_ECDSA_C
```

Comment:
Meant to reply to this much earlier and got stuck into something else and distracted. Apologies.

So my understanding goes like this:
- We need to enable `MBEDTLS_PSA_CRYPTO_CONFIG` since this will default to being enabled in 4.0.
- We are then only replacing legacy config options where a suitable PSA equivalent exists, in this case there is not a PSA way of enabling everest.
- We also leave legacy context lines where the options refer to something other than crypto, for example TLS, as is the case with options like `MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED`.

I am not 100% sure I have answered your question precisely, let me know!

Reply:
A good way to convince yourself about what options are needed or not is to use the program `query_compile_time_config`. Invoking the program with `-l` will list all the enabled options at compile time. If you run this on the old component, and then again on the new component (as modified by this PR), the two lists should be identical (with a couple of trivial exceptions). This is partly how I have been determining what is needed and what isn't.